compiler/rustc_mir_build/src/build/mod.rs

```diff
@@ -403,6 +403,21 @@ enum NeedsTemporary {
 #[must_use = "if you don't use one of these results, you're leaving a dangling edge"]
 struct BlockAnd<T>(BasicBlock, T);
 
+impl BlockAnd<()> {
+    /// Unpacks `BlockAnd<()>` into a [`BasicBlock`].
+    #[must_use]
+    fn unpack_block(self) -> BasicBlock {
+        let Self(block, ()) = self;
+        block
+    }
+
+    /// Explicitly discards a [`BlockAnd<()>`] value.
+    ///
+    /// This is similar to writing `let _ = ...`, except that it is also
+    /// (a) postfix, and (b) searchable.
+    fn unpack_discard(self) {}
```

Comment:
It's not really unpacking anything. I'm not super convinced by this method, I'd just use `let _ = …` as the docs mention but er, it's fine.